src/neptune/internal/utils/git.py

```diff
@@ -90,3 +107,95 @@ def to_git_info(git_ref: Union[GitRef, GitRefDisabled]) -> Optional[GitInfo]:
         )
     except:  # noqa: E722
         return None
+
+
+class DiffTracker:
+    def __init__(self, repo: git.Repo):
+        self._repo = repo
+        self._head = self._repo.head
```

Comment:
Could we add type hints here (`HEAD` and `Optional[str]` for `self._upstream_commit_sha`)?